### PR TITLE
Add Win32-network as separate repository

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

**Review by rvl — COMMENTED** [review]
The stack build in CI is failing because of unsupported cabal version.

The cabal version _could_ be increased, but do you absolutely need 3.0 features in that package?

The use of `common` stanzas in [`Win32-network.cabal`](https://github.com/input-output-hk/Win32-network/blob/main/Win32-network.cabal#L1) seems gratuitous bordering on incorrect.

**Comment by newhoggy** [comment]
Fortunately I didn't use anything from `3.0`.

File: snapshots/cardano-1.25.1-GHC-8.10.4.yaml [hunk 1]
```diff
@@ -108,12 +108,14 @@ packages:
...
   - typed-protocols
   - typed-protocols-examples
-  - Win32-network
   # Extra packages not used by cardano-node
   - cardano-client
   - ntp-client
   - ouroboros-consensus-mock
 
+- git: https://github.com/input-output-hk/Win32-network
+  commit: f8a1bcd08608d78e200f600b5e9d2cd9dec233f2
```
Comment (rvl) [inline snapshots/cardano-1.25.1-GHC-8.10.4.yaml:117]: This commit is different to the one in the other file. I think f71be82 is the correct rev.

Comment (newhoggy) [reply snapshots/cardano-1.25.1-GHC-8.10.4.yaml]: Good spot.  Thanks!

**Review by rvl — APPROVED** [review]

**Pull request merged**